### PR TITLE
Stop time validator

### PR DESCRIPTION
**Pull request by lilian-hunt**

### CODE REVIEW

**Review by aannie-y — COMMENTED** [review]
lgtm

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.List;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.Inject;
+import org.mobilitydata.gtfsvalidator.notice.DecreasingStopTimeDistanceNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+
+/**
+ * Use case to validate that for each trip, stop times have increasing `shape_dist_travelled` values.
```
Comment (KathleenQ) [inline gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java]: Directly write "Validates that ...." as the start, to be consistent with the beginning explanations for all validator rules.

**Review by KathleenQ — COMMENTED** [review]
Except for the comment that I mentioned, LGTM

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java (hunk 1 above)
Comment (lilian-hunt) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java]: Updated.

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    // Create stopTimeTable:
+    final List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(stopTime1);
+    stopTimes.add(stopTime2);
+    stopTimes.add(stopTime3);
+    validator.table = GtfsStopTimeTableContainer.forEntities(stopTimes, noticeContainer);
+
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices())
+        .containsExactly(new DecreasingStopTimeDistanceNotice("trip1", 3, 3, 1.4, 2, 2, 1.5));
```
Comment (jupi3200) [inline gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java]: Please either update DecreasingStopTimeDistanceNotice to have a builder and use it, or improve readability by commenting at the call site to indicate which param you are setting.

For example:
DecreasingStopTimeDistanceNotice(/* tripId= */ "trip1", /* csvRowNumber = */ 3, /* stopSequence = */ 3, /* shapeDistTraveled = */ 1.4, /* prevCsvRowNumber= */ 2, /* prevStopSequence= */ 2, /* prevShapeDistTraveled = */ 1.5)

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+
+@RunWith(JUnit4.class)
+public class StopTimeIncreasingDistanceValidatorTest {
+  private final GtfsStopTime stopTime1 = new GtfsStopTime.Builder()
+                                             .setCsvRowNumber(1)
+                                             .setTripId("trip1")
+                                             .setStopId("1001")
+                                             .setStopSequence(1)
```
Comment (jupi3200) [inline gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java]: The test might be somewhat more likely to detect some errors is you avoid having stopSequence == CsvRowNumber

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                                             .setShapeDistTraveled(1.0)
+                                             .build();
+  private final GtfsStopTime stopTime2 = new GtfsStopTime.Builder()
+                                             .setCsvRowNumber(2)
+                                             .setTripId("trip1")
+                                             .setStopId("1002")
+                                             .setStopSequence(2)
+                                             .setShapeDistTraveled(1.5)
+                                             .build();
+  private final GtfsStopTime stopTime3 = new GtfsStopTime.Builder()
```
Comment (jupi3200) [inline gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java]: perhaps the test will be more readable if the variable would be named according to the intended usage.
For example I would have named this as stop_3_at_1_5

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java (hunk 2 above)
Comment (lilian-hunt) [reply gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java]: Updated to comment at the call site

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java (hunk 3 above)
Comment (lilian-hunt) [reply gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java]: Changed the csvRowNumber to be random now, stop name corresponds to its sequence not csv number now

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java (hunk 4 above)
Comment (lilian-hunt) [reply gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java]: Good point, have updated now

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            && prev.shapeDistTraveled() > curr.shapeDistTraveled()) {
+          noticeContainer.addNotice(
+              new DecreasingStopTimeDistanceNotice(
+                  curr.tripId(),
+                  curr.csvRowNumber(),
+                  curr.stopSequence(),
+                  curr.shapeDistTraveled(),
+                  prev.csvRowNumber(),
+                  prev.stopSequence(),
+                  prev.shapeDistTraveled()));
```
Comment (jupi3200) [inline gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:56]: Should we have multiple notices per trip?

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java (hunk 3 above)
Comment (jupi3200) [reply gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java]: Did you commit the change?

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java (hunk 5 above)
Comment (lilian-hunt) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:56]: Yes, I think if this issue happens more than once there should be multiple notices

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java (hunk 3 above)
Comment (lilian-hunt) [reply gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java]: Yes I have. I can see it and this file is tagged "outdated"

Comment (lilian-hunt) [reply gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidatorTest.java]: Screen Shot 2021-01-28 at 11 10 47 am
What I can see

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java (hunk 5 above)
Comment (jupi3200) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:56]: Should it?

Comment (lilian-hunt) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java:56]: Well yes, if the shape distance travelled is decreasing at other row numbers, the user needs to know all row numbers at which this occurs. So they can address the issue at all places

**Review by jupi3200 — APPROVED** [review]

**Pull request merged**